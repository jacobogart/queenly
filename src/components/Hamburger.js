import React from "react";

class Hamburger extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showMenu: false
    };
  }

  toggleBurger = () => {
    this.state.showMenu
      ? this.setState({ showMenu: false })
      : this.setState({ showMenu: true });
  };

  handleMouseOut = () => {
    this.setState({ showMenu: false });
  };

  clickHamburger = event => {
    let type = event.target.value.split(" ")[2];
    this.props.displayAllOfType(type);
    this.props.displayAllSearchResults();
  };

  render() {
    if (this.state.showMenu) {
      return (
        <div className="hamburgerMenu">
          <i className="fas fa-bars fa-2x" onClick={this.toggleBurger} />
          <div className="burger-container" onMouseLeave={this.handleMouseOut}>
            <ul className="showMenu" onClick={this.clickHamburger}>
              <li>
                <input type="button" className="menu" value="Show All Venues" />
              </li>
              <li>
                <input type="button" className="menu" value="Show All Shows" />
              </li>
              <li>
                <input type="button" className="menu" value="Show All Queens" />
              </li>
              <li>
                <input
                  type="button"
                  className="menu"
                  value="Show All Favorites"
                />
              </li>
            </ul>
          </div>
        </div>
      );
    } else {
      return (
        <div className="hamburgerMenu">
          <i className="fas fa-bars fa-2x" onClick={this.toggleBurger} />
        </div>
      );
    }
  }
}

export default Hamburger;
