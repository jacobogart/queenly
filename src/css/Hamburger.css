.showMenu {
  background-color: #202828;
  position: absolute;
  z-index: 100;
  list-style: none;
  left: 10px;
}

.hamburgerMenu {
  margin: 0 0 0 8px;
}

.hamburgerMenu ul {
  padding: 0;
}

.showMenu input {
  text-align: left;
  background-color: #202828;
  color: lightgray;
  font-size: 24px;
  border: none;
  padding: 5px 20px 10px;
  width: 100%;
}

.showMenu input:hover {
  background-color: white;
  color: #202828;
}