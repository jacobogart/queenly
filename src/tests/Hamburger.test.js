import React from "react";
import Hamburger from "../components/Hamburger";
import { shallow } from "enzyme";

const mockDisplayAllOfType = jest.fn();
const mockDisplayAllSearchResults = jest.fn();

describe("Hamburger", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Hamburger
        displayAllOfType={mockDisplayAllOfType}
        displayAllSearchResults={mockDisplayAllSearchResults}
      />
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have default state", () => {
    expect(wrapper.state()).toEqual({ showMenu: false });
  });

  it("should toggle showMenu when toggleBurger is invoked", () => {
    expect(wrapper.state()).toEqual({ showMenu: false });
    wrapper.instance().toggleBurger();
    expect(wrapper.state()).toEqual({ showMenu: true });
    wrapper.instance().toggleBurger();
    expect(wrapper.state()).toEqual({ showMenu: false });
  });

  it("should set showMenu to false when handleMouseOut is invoked", () => {
    expect(wrapper.state()).toEqual({ showMenu: false });
    wrapper.instance().toggleBurger();
    expect(wrapper.state()).toEqual({ showMenu: true });
    wrapper.instance().handleMouseOut();
    expect(wrapper.state()).toEqual({ showMenu: false });
  });

  it("should invoke displayAllOfType and displayAllSearchResults when clickHamburger is invoked", () => {
    const mockEvent = {target: {value: "Show All Queens"}};
    wrapper.instance().clickHamburger(mockEvent);
    expect(mockDisplayAllOfType).toHaveBeenCalledWith("Queens");
    expect(mockDisplayAllSearchResults).toHaveBeenCalled();
  });
});