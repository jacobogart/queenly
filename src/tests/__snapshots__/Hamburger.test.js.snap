// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger should match snapshot 1`] = `
<div
  className="hamburgerMenu"
>
  <i
    className="fas fa-bars fa-2x"
    onClick={[Function]}
  />
</div>
`;
